backends/dpdk/dpdkHelpers.cpp

```diff
@@ -1030,6 +1055,52 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
             if (a->method->getName().name != "execute") {
                 BUG("Meter function not implemented.");
             }
+        } else if (a->originalExternType->getName().name == "DirectMeter") {
+            if (a->method->getName().name != "execute") {
+                BUG("Direct Meter function %1% not implemented.", a->method->getName().name);
+            }
+        } else if (a->originalExternType->getName().name == "DirectCounter") {
+            auto di = a->object->to<IR::Declaration_Instance>();
+            auto declArgs = di->arguments;
+            unsigned value = 0;
+            auto counter_type = declArgs->at(0)->expression;
+            if (counter_type->is<IR::Constant>())
```

Comment:
usually you write if (auto c = counter_type->to<IR::Constant>()) value = c->asUnsigned().